lib/portlayer/storage/vsphere/image.go

```diff
@@ -283,6 +246,124 @@ func (v *ImageStore) WriteImage(ctx context.Context, parent *portlayer.Image, ID
 	return newImage, nil
 }
 
+// Create a temporary directory, create a vmdk in this directory, attach/mount
+// the disk, unpack the tar, check the checksum.  If the data doesn't match the
+// expected checksum, abort by nuking the tempdir.  If everything matches, move
+// the tmpdir to the expected location (with the vmdk inside it).  The unwind
+// path is a bit convoluted here;  we need to clean up on the way out in the
+// error case (using the tmpdir).
+func (v *ImageStore) writeImage(ctx context.Context, storeName, parentID, ID,
+	sum string, r io.Reader) error {
+
+	// Create a temp image directory in the store.
+	tmpImageDir := v.imageDirPath(storeName, ID+"inprogress")
+	_, err := v.ds.Mkdir(ctx, true, tmpImageDir)
+	if err != nil {
```

Comment:
The only way for the `ID+inprogress` directory to exist after a `writeImage` call is when the VCH is rebooted in the middle of `writeImage`.  And in that case the image will be added to the image list and one could `rmi` it.